lib/etcdec.cxx

```diff
@@ -174,6 +174,10 @@ submitted to the exclusive jurisdiction of the Swedish Courts.
 #if defined(_MSC_VER)
```

Comment:
I could do that yeah. An easy refactor for sure.
ClangCL does report MSVC's error codes so it does also handle the MSVC specific warning statements. That's why I did it this way, I figured whatever warnings we want to disable for MSVC we also want to disable for ClangCL.

Reply:
It looks like you did refactor it.

If CLangCL is handling MSVC specific warnings there shouldn't be any need for the `#pragma clang diagnostic ignored "-Wunused-parameter"` as that is the same as warning 4100 in CL. Because we do need it, it looks like at best `#pragma warning` is being ignored. Is CLangCL's handling of `#pragma warning` documented somewhere?

I agree with the statement "I figured whatever warnings we want to disable for MSVC we also want to disable for ClangCL.". If the original way you did it implements this then restore it but add a comment as to why and about the way CLangCL handles these things.

Reply:
Yes sorry looks like it just ignores them. This current approach works the same as it did before, it's no longer if/else with `_MSC_VER` so we apply msvc style warning pragmas for any compiler that has it defined. It's a little odd to read though so if you want I can make it explicit like I did before

Reply:
The way you have it is fine but I suggest adding a comment ` // clang_cl defines both _MSC_VER and __clang__` to prevent somebody going down the road of trying `#elif` in future.